Packages/com.unity.inputsystem/InputSystem/Plugins/iOS/iOSSensors.cs

```diff
@@ -0,0 +1,25 @@
+#if UNITY_EDITOR || UNITY_IOS || UNITY_TVOS
+using System.Runtime.InteropServices;
+using UnityEngine.Experimental.Input.LowLevel;
+using UnityEngine.Experimental.Input.Plugins.iOS.LowLevel;
+using UnityEngine.Experimental.Input.Utilities;
+
+namespace UnityEngine.Experimental.Input.Plugins.iOS.LowLevel
+{
+    [StructLayout(LayoutKind.Explicit, Size = 52)]
+    public unsafe struct MotionDeviceState : IInputStateTypeInfo
+    {
+        public static FourCC kFormat = new FourCC('I', 'M', 'T', 'N');
+
+        [InputControl][FieldOffset(0)] public Vector3 gravity;
+        [InputControl][FieldOffset(12)] public Quaternion attitude;
+        [InputControl][FieldOffset(28)] public Vector3 acceleration;
+        [InputControl][FieldOffset(40)] public Vector3 rotationRateUnbiased;
+
+        public FourCC GetFormat()
+        {
+            return kFormat;
+        }
+    }
```

Comment:
So just to confirm, how should I approach this, you seems to agreed that we should create artifical sensors for each of those properties, right?

Reply:
I didn't come up with a good solution on the managed side. ATM my thinking is that the iOS side should split up its motion sensor code path such that it registers and feeds several sensors from the motion data.

There probably is a more elegant solution somewhere in here but ATM, having the backend and the C# system develop different views on the device setup and on where to route state would require some more invasive changes.